cmake/windows-macros.cmake

```diff
@@ -0,0 +1,131 @@
+#-------------------------------------------------------------------------------
+# _detach_debuginfo(<target> <dest>)
+#
+# Helper macro to detach the debug information from the target.
+#
+# The debug info of the given target are extracted and saved on a different file
+# with the extension '.dbg', that is installed alongside the target in the .debug folder.
+#-------------------------------------------------------------------------------
+macro(_detach_debuginfo target dest)
+    set (CMAKE_OBJCOPY, objcopy)
+
+    add_custom_command (TARGET ${target} POST_BUILD
+      COMMAND ${CMAKE_COMMAND} -E make_directory .debug
+      COMMAND objcopy --only-keep-debug $<TARGET_FILE:${target}> $<TARGET_FILE_DIR:${target}>/$<TARGET_FILE_NAME:${target}>.dbg
+      COMMAND objcopy --strip-debug $<TARGET_FILE:${target}>
+      COMMAND objcopy --add-gnu-debuglink=$<TARGET_FILE_NAME:${target}>.dbg $<TARGET_FILE:${target}>
+      COMMENT "Detaching debug infos for ${target}."
+    )
+
+    # ensure that the debug file is installed on 'make install'...
+    install(
+      FILES $<TARGET_FILE_DIR:${target}>/$<TARGET_FILE_NAME:${target}>.dbg
+      DESTINATION ${dest}/.debug
+      COMPONENT DTDebugSymbols)
+
+    # ... and removed on 'make clean'.
+    set_property(DIRECTORY APPEND PROPERTY ADDITIONAL_MAKE_CLEAN_FILES $<TARGET_FILE_NAME:${target}>.dbg)
+endmacro()
+
+function(InstallDependencyFiles)
+
+if (WIN32)
+  # Dependency files (files which needs to be installed alongside the darktable binaries)
+  # must be in the bin directory
+  message( STATUS "WIN32: Adding dependency files to install" )
+  get_filename_component(MINGW_PATH ${CMAKE_CXX_COMPILER} PATH )
+
+  set( CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS
+  #GTK3
```

Comment:
Indeed a lot of dependencies were missing. I have added many of them (see next commit), but still: for GraphicsMagick, libgphoto2, gdk-pixbuf, adwaita-icon-theme I need to find a way to install additional files and config files which those libraries are using. Unfortunately I'm not aware of any easy way to install the automatically with the Windows installer, its not so smart like the package managers in unix.